dpnp/backend/kernels/dpnp_krnl_arraycreation.cpp

```diff
@@ -110,6 +110,25 @@ void dpnp_full_like_c(void* array_in, void* result, const size_t size)
     dpnp_full_c<_DataType>(array_in, result, size);
 }
 
+template <typename _DataType>
+void dpnp_identity_c(void* result1, const size_t n)
+{
+    _DataType* result = reinterpret_cast<_DataType*>(result1);   
+
+    for (size_t i = 0; i < n; ++i)
+    {
+        for (size_t j = 0; j < n; ++j)
+        {
+            if (i == j)
+            {
+                result[i * n + j] = 1;
+            } else {
+                result[i * n + j] = 0;
+            }
+        }
+    }
```

Comment:
We could parallelize it with SYCL if we rolled up nested loops to a single one as below
```suggestion
    for (size_t i = 0; i < n * n; ++i)
    {
        if (i / n == i % n)
        {
            result[i] = 1;
        }
        else
        {
            result[i] = 0;
        }
    }
```

Reply:
We can use 2-dim range here if it is needed.

Reply:
kernel might look like this:
```cpp
    cl::sycl::range<2> gws(size, size);
    auto kernel_parallel_for_func = [=](cl::sycl::id<2> global_id) {
        size_t i = global_id[0];
        size_t j = global_id[1];
        result[i * n + j] = (i == j) ? 1 : 0;
    };
```

But before this you need check on `size`. It should be non-zero.